apps/prairielearn/src/sync/fromDisk/assessments.ts

```diff
@@ -95,33 +96,37 @@ function getParamsForAssessment(
 
   let alternativeGroupNumber = 0;
   let assessmentQuestionNumber = 0;
-  const allRoleNames = (assessment.groupRoles ?? []).map((role) => role.name);
+  const allRoleNames = new Set((assessment.groupRoles ?? []).map((role) => role.name));
   const assessmentCanView = assessment?.canView ?? allRoleNames;
   const assessmentCanSubmit = assessment?.canSubmit ?? allRoleNames;
   const alternativeGroups = (assessment.zones ?? []).map((zone) => {
-    const zoneGradeRateMinutes = _.has(zone, 'gradeRateMinutes')
-      ? zone.gradeRateMinutes
-      : assessment.gradeRateMinutes || 0;
+    const zoneGradeRateMinutes =
+      Object.prototype.hasOwnProperty.call(zone, 'gradeRateMinutes') &&
+      zone.gradeRateMinutes !== undefined
+        ? zone.gradeRateMinutes
+        : assessment.gradeRateMinutes || 0;
     const zoneCanView = zone?.canView ?? assessmentCanView;
     const zoneCanSubmit = zone?.canSubmit ?? assessmentCanSubmit;
     return zone.questions.map((question) => {
       let alternatives: {
         qid: string;
-        maxPoints: number | number[];
-        points: number | number[];
-        maxAutoPoints: number | number[];
-        autoPoints: number | number[];
-        manualPoints: number;
+        maxPoints: number | null;
+        points: number | number[] | null;
+        maxAutoPoints: number | null;
+        autoPoints: number | number[] | null;
+        manualPoints: number | null;
         forceMaxPoints: boolean;
         triesPerVariant: number;
         gradeRateMinutes: number;
-        canView: string[] | null;
-        canSubmit: string[] | null;
+        canView: Set<string> | null;
+        canSubmit: Set<string> | null;
```

Comment:
How does this work with these being sets? All these objects are ultimately serialized with `JSON.stringify`, and I can't see how that could work:

```
> JSON.stringify({foo: new Set(["1", "2"])})
'{"foo":{}}'
```

Reply:
Fixed in [0756a14](https://github.com/PrairieLearn/PrairieLearn/pull/11024/commits/0756a147abffc99fa3065ee0de2a2f1e52b1056d).

Reply:
Let's continue discussion here: https://github.com/PrairieLearn/PrairieLearn/pull/11024#discussion_r1907917205